src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -232,6 +240,30 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
                             rawResource = await streamReader.ReadToEndAsync();
                         }
 
+                        // if we have more columns, it means sort expressions were added.
+                        if (reader.FieldCount > 9 && searchOptions.Sort != null && searchOptions.Sort.Count > 0)
+                        {
+                            SortOrder sortOrder = searchOptions.Sort[0].sortOrder;
+
+                            // if sort is requested asc keep the first result for continuation token
```

Comment:
Are you using the first match? That will not have the right value for continuation.